src/openttd.cpp

```diff
@@ -939,7 +940,17 @@ bool SafeLoad(const std::string &filename, SaveLoadOperation fop, DetailedFileTy
 	_game_mode = newgm;
 
 	switch (lf == nullptr ? SaveOrLoad(filename, fop, dft, subdir) : LoadWithFilter(lf)) {
-		case SL_OK: return true;
+		case SL_OK:
+			/* Start the GameScript. */
+			Game::StartNew();
```

Comment:
Does the GS/AI start need to check _networking and _network_server here? Or is that already done somewhere else?

I would have expected the GS/AI start at the end of "AfterLoadGame" together with "InitializeWindowsAndCaches"/"AfterLoadLinkGraphs". I am not sure whether this place is equivalent.

Reply:
`StartNew()` checks `_networking` and `_network_server`.

I tried to start in "AfterLoadGame", was not a good idea as it also started them in intro game (and for dedicated servers it would start them at the end of test SaveOrLoad, then again after the real load done by `SM_LOAD_GAME` mode switch), while `SafeLoad()` is only done when switching to `SM_LOAD_GAME`

Reply:
After working on #10241, I now know scripts must be started at the end of "AfterLoadGame".